### ISSUE TITLE
Contrib data types cause type error (metatag issue)

### ISSUE THREAD

**Issue by pmelab** [drupal-graphql/graphql#609]
If a module (e.g. `metatag`) introduces a new primitive data type, it is not part of the derived types, but any field using it will reference it. That results in a "Missing type metatag." exception.

**Comment by pmelab** [comment]
The current workaround is to create a custom scalar:

```php
<?php

namespace Drupal\custom\Plugin\GraphQL\Scalars;

use Drupal\graphql\Plugin\GraphQL\Scalars\Internal\StringScalar;

/**
 * Metatag module dummy type.
 *
 * Metatag module defines a custom data type that essentially is a string, but
 * not called string. And the GraphQL type system chokes on that.
 *
 * @GraphQLScalar(
 *   id = "metatag",
 *   name = "metatag",
 *   type = "string"
 * )
 */
class MetatagScalar extends StringScalar {

}
```

**Comment by bappa-ww** [comment]
I am facing wsod issue at url /graphql/explorer after I installed metatag module
```
Uncaught TypeError: Cannot read property 'types' of undefined
    at i (bundle.min.js?v=8.5.3:1)
    at Object.attach (bundle.min.js?v=8.5.3:1)
    at drupal.js?v=8.5.3:25
    at Array.forEach (<anonymous>)
    at Object.Drupal.attachBehaviors (drupal.js?v=8.5.3:22)
    at drupal.init.js?v=8.5.3:16
    at HTMLDocument.t (ready.min.js?v=1.0.8:4)
```

**Comment by bappa-ww** [comment]
@pmelab Fixed the issue by adding the custom scalar for the metatag as you mentioned! Thanks!

**Comment by weitzman** [comment]
same error here ... shall we ask the metatag module add this plugin to its codebase?

**Comment by damienmckenna** [comment]
Out of interest, does this fix the problem too? https://www.drupal.org/project/metatag/issues/2977545

**Comment by damienmckenna** [comment]
Ah! The patch in 2977545 is pmelab's code above, thank you for that.

**Comment by damienmckenna** [comment]
I've committed @pmelab's change (via @bappa-ww) to the Metatag module, it'll be in the next release. Feel free to reach out directly, or post in the module's issue queue, if you find any other problems or have suggestions. Thank you!

**Comment by pmelab** [comment]
@damienmckenna thanks for fixing this in metatag!

But I think we need a more generic solution. Primitives are a valid extension point of typed data and we can't force every other module to think about GraphQL types.

Perhaps just auto-add a String-like type for every unknown primitive?

**Comment by bappa-ww** [comment]
@pmelab I agree to add a String type for any unknown primitive by default!

**Comment by fubhy** [comment]
Solved in #665
